### ISSUE TITLE
Looking for suggestions on training customized dataset

### ISSUE THREAD

**Issue by triangleCZH** [rpautrat/SuperPoint#197]
Hi, thank you very much for your great work!
I want to transfer the COCO pretrain to my own satellite images, which are high resolution bird views.
I first try the pretrain weight provided by MagicLeap (superpoint_v1.pth), and find out it already has good results for NN+ransac. Therefore I directly export the label with pretrain and start the training of superpoint.

However, I encounter problems (1) compared to the pretrained weight, the predictions of newly trained model has lower confidence, the details are shown in heatmap_pretrain and heatmap_trained. You can see that the new heatmap gets blur and not so confident. (2) and the white area around each keypoint seems to be too large, which we would expect it to get smaller into a bright point

heatmap_pretrain
![heatmap1](https://user-images.githubusercontent.com/20054340/111906256-0c9df180-8a26-11eb-910c-c80299d0a966.jpg)
heatmap_trained
![heatmap2](https://user-images.githubusercontent.com/20054340/111906260-10ca0f00-8a26-11eb-9777-492c12be789a.jpg)

**Comment by triangleCZH** [comment]
My first question will be, would you think it is ok to (a)directly start with superpoint training on a new dataset, or maybe (b)disable the descriptor loss and train detector first? I train directly on superpoint because I observe good result on the pretrain weights, but they seem to be ruined during training. I do get better result when I disable the descriptor loss, but I am not sure whether it is appropriate.
If the answer to the first question is (b), I have also tried to keep train on COCO and encounter similar problem (not as confident as pretrain, not bright and large greys). Given that pretrain is from COCO, I would expect it feasible to start with superpoint training, but the result still disagrees.

Please let me know if I did not make anything clear, and I am looking forward to your reply.

**Comment by triangleCZH** [comment]
Some details of my training:

I crop high resolution images into 360 * 360 data, and export the label with 100 HAs on the superpoint_v1.pth. 
Then I enable the warp loss and the descriptor loss and train for 10 epochs, which is around 40k iters. I follow the parameter settings in your config, the learning rate was 0.0001 (0.001 does not converge).

**Comment by rpautrat** [comment]
Hi,

In general, the detections of SuperPoint are a bit more blurry than when training the detector alone. On COCO, I didn't find necessary to pre-train with the detector alone and then fine-tune with the descriptor part. It was enough to directly train SuperPoint from scratch. But for satellite images, it might help to do this pre-training with the detector only.

But overall, I would not evaluate the performance of your trained model just by looking at the confidence map and checking if it is blurry or with lower confidence. This is hardly comparable, and the NMS should anyway dampen the effect of the blurriness. I recommend sticking with the usual keypoint metrics like repeatability and localization error rather.

**Comment by triangleCZH** [comment]
@rpautrat  Thank you for your reply! I will try the metrics you mentioned, and will update in this post if I get anything new
